src/ansys_sphinx_theme/navbar_dropdown.py

```diff
@@ -0,0 +1,182 @@
+'caption', and 'sections' keys. The 'sections' key contains a list of
+sub-entries, allowing for nested navigation structures.
+"""
+
+
+def update_template_context(app, pagename, templatename, context, doctree):
+    """Inject custom variables and utilities into the Sphinx template context."""
+
+    @lru_cache(maxsize=None)
+    def render_navbar_links_html() -> bs4.BeautifulSoup:
+        """Render external header links as HTML for the navbar."""
+        if not hasattr(app.config, "navbar_contents"):
+            raise ValueError("Navbar configuration is missing. Please specify a navbar YAML file.")
+        node = nodes.container(classes=["navbar-content"])
+        node.append(build_navbar_nodes(app.config.navbar_contents))
+        header_soup = bs4.BeautifulSoup(app.builder.render_partial(node)["fragment"], "html.parser")
+        return add_navbar_chevrons(header_soup)
+
+    def build_navbar_nodes(obj: List[NavEntry], is_top_level: bool = True) -> nodes.Node:
+        """Recursively build navbar nodes from configuration entries."""
+        bullet_list = nodes.bullet_list(
+            bullet="-",
+            classes=["navbar-toplevel" if is_top_level else "navbar-sublevel"],
+        )
+        for item in obj:
+            if "file" in item:
+                ref_node = make_refnode(
+                    app.builder,
+                    context["current_page_name"],
+                    item["file"],
+                    None,
+                    nodes.inline(classes=["navbar-link-title"], text=item.get("title")),
+                    item.get("title"),
+                )
+            elif "link" in item:
+                ref_node = nodes.reference("", "", internal=False)
+                ref_node["refuri"] = item.get("link")
+                ref_node["reftitle"] = item.get("title")
+                ref_node.append(nodes.inline(classes=["navbar-link-title"], text=item.get("title")))
+            else:
+                continue
+            if "caption" in item:
+                caption = nodes.Text(item.get("caption"))
+                ref_node.append(caption)
+            paragraph = nodes.paragraph()
+            paragraph.append(ref_node)
+            container = nodes.container(classes=["ref-container"])
+            container.append(paragraph)
+            list_item = nodes.list_item(
+                classes=["active-link"] if item.get("file") == pagename else []
+            )
+            list_item.append(container)
+            if "sections" in item:
+                wrapper = nodes.container(classes=["navbar-dropdown"])
+                wrapper.append(build_navbar_nodes(item["sections"], is_top_level=False))
+                list_item.append(wrapper)
+            bullet_list.append(list_item)
+        return bullet_list
+
+    context["render_navbar_links_html"] = render_navbar_links_html
+    context["pygments_highlight_python"] = lambda code: highlight(
+        code, PythonLexer(), HtmlFormatter()
+    )
```

Comment:
Probably it is good to refactor this, it does not look like it belongs to the above code.

Reply:
done